kobo/xmlrpc.py

```diff
@@ -479,7 +480,8 @@ def request(self, *args, **kwargs):
                 except KeyboardInterrupt:
                     raise
                 except (socket.error, socket.herror, socket.gaierror, socket.timeout,
-                        httplib.CannotSendRequest, xmlrpclib.ProtocolError) as ex:
+                        httplib.CannotSendRequest, xmlrpclib.ProtocolError,
+                        PermissionDenied) as ex:
```

Comment:
Are you sure about the exception?
It looks that it will actually raise something else:
```
xmlrpc.client.Fault: <Fault 1: PermissionDenied: Login required.>
Frame close in /usr/lib64/python3.11/xmlrpc/client.py at line 668
<CODE>
     661         self._use_bytes = use_builtin_types
     662 
     663     def close(self):
     664         # return response tuple and target method
     665         if self._type is None or self._marks:
     666             raise ResponseError()
     667         if self._type == "fault":
-->  668             raise Fault(**self._stack[0])
     669         return tuple(self._stack)
     670 
     671     def getmethodname(self):
     672         return self._methodname
     673 
     674     #
</CODE>
```

Reply:
Also I am thinking about - if we should rather retry it on hub side.

Reply:
I think it'd be better to keep it in Kobo, so the retry logic can be kept unified. 

I'd agree with having the logic in hub if we were going to do something between retries (i.e. trigger login again)